include/real/real.hpp

```diff
@@ -66,6 +66,7 @@ namespace boost {
             real(std::shared_ptr<real_data<T>> x) : _real_p(x){};
 
         public:
+            using exponent_t = int;
```

Comment:
Can you move it to the template description and default it to int? 
Also, it may make sense to start calling T something more specific.